client/consul_template.go

```diff
@@ -26,17 +26,17 @@ var (
 // TaskHooks is an interface which provides hooks into the tasks life-cycle
 type TaskHooks interface {
 	// Restart is used to restart the task
-	Restart()
+	Restart(source, reason string)
```

Comment:
Idea was source may be different plugins and reason is for them to inform why for understanding why the task gets restarted